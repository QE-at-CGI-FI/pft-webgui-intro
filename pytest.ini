[pytest]
addopts = --browser=firefox
          --html=test-results/report.html