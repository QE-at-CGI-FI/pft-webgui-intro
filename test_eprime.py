import requests
import approvaltests
from playwright.sync_api import Page
import pytest

URL = "https://qe-at-cgi-fi.github.io/eprime/"

def test_request():
    response = requests.get(URL)
    assert response.status_code == 200
    approvaltests.verify_html(response.text)

def this_is_sample():
    with open("sample.txt") as f:
        lines = f.readlines()
    return str(lines)

data = [
    ("to be or not be", "5", "2", "0"), 
    ("maaret's", 1, 0, 1), 
    ("to be " *1000, 2000, 1000, 0), 
    (this_is_sample(), 2, 0, 0),
    #bug: too many words
    #("to be or not to be - hamlet's dilemma", 8, 2, 1)
]
ids=["hamlet", "maaret", "long text", "file"]


@pytest.mark.parametrize("input_text, expect_1, expect_2, expect_3", data, ids=ids)
def test_playwright(page: Page, input_text, expect_1, expect_2, expect_3):
    page.goto(URL)
    page.fill("#test_field", input_text)
    page.click("#test_button")
    assert page.inner_text("#test_number_1") == str(expect_1)
    assert page.inner_text("#test_number_2") == str(expect_2)
    assert page.inner_text("#test_number_3") == str(expect_3)
    


# SESSION
# Input from file

# SESSION
# Run in GitHub Actions
# Fix the asserts to expect -style
# Refactor to classes and page objects
# pytest.ini and html report
# Fixtures for webgui in conftest.py
# Allure report
