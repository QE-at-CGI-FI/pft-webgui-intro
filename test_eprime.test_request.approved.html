<!DOCTYPE html>
<html lang="en">
 <head>
  <title>
   Eprimer - e-prime checking tool
  </title>
  <script src="eprime.js">
  </script>
  <link href="https://fonts.googleapis.com/css?family=Lato|Lobster" rel="stylesheet"/>
  <link href="style.css" rel="stylesheet"/>
 </head>
 <body>
  <div class="page">
   <img src="./eviltester.png" width="100%"/>
   <br/>
   <code>
    This test target is from collections of
    <a href="https://github.com/eviltester/TestingApp">
     Alan Richardson, eviltester
    </a>
    , a brilliant exploratory tester.
   </code>
   <div class="listpageintro">
    <h1>
     E-Primer an e-prime checking tool
    </h1>
    <p>
     Do you want to write without using the verb "to be"?
    </p>
    <p>
     Do you want to master
     <a href="http://en.wikipedia.org/wiki/E-Prime" rel="nofollow" target="_blank">
      e-prime
     </a>
     ?
    </p>
    <p>
     Use our online tool to check your writing.
    </p>
   </div>
   <div id="app">
    <div id="statistics">
     <ul>
      <li>
       Word Count:
       <span id="test_number_1">
       </span>
      </li>
      <li>
       Discouraged Words:
       <span id="test_number_2">
       </span>
      </li>
      <li>
       Possible Violations:
       <span id="test_number_3">
       </span>
      </li>
     </ul>
    </div>
    <div class="eprime_text" id="eprimeoutput">
    </div>
    <label for="eprime_text">
     Text:
    </label>
    <br>
     <textarea cols="60" id="test_field" rows="5"></textarea>
     <br/>
     <button id="test_button" onclick="processInputText();">
      Check For E-Prime
     </button>
    </br>
   </div>
  </div>
 </body>
</html>
